tb_plugin/torch_tb_profiler/profiler/data.py

```diff
@@ -181,9 +187,28 @@ def analyze(self):
                    )
             self.recommendations.append(text)
 
-        self.analyze_gpu_metrics()
+        if self.use_dp and len(self.used_devices) > 1:
```

Comment:
better to move these logic into a separate helper method like gpu metrics

Reply:
done